handbook/engineering/code-insights/index.md

```diff
@@ -0,0 +1,34 @@
+# Code insights team (coming soon)
```

Comment:
My personal preference is to not prematurely create a team page like this until the team exists. The team can't exist until we have a PM. Can we inline this into the web team?

Reply:
This information (at least the goals and mission statement) is currently inlined in the web team page, but @jeanduplessis and I were worried that it is burried too deep in there for a candidate (who came to the job listing) to find it and can't do a good job to get them excited specifically for this team (because they for example have an interest in this mission specifically, or bring unique experience and skills for it).

The next step would be to link to this page directly from the "These teams are hiring for this role" section in the job listings, instead of just listing "Web". I am worried that "Web" isn't enticing to someone who doesn't want to join a (perceived) general-purpose frontend team - but 
 our split into dedicated product (and platform) teams is right around the corner.
And in the context of the backend engineer job listing, "Web" is very confusing to list (it's not enticing from the name, and a candidate would have to read between the lines of the web team page to imagine what they would even be working on).

Reply:
@nicksnyder I had direct feedback from a candidate that they didn't really understand what the Code Insights team was about when they tried preparing for their interview. I'm not against inlining it in the Web team page, but as Felix mentioned, that page can quickly start getting busy and we need to do a better job of selling the Code Insights team for possible candidates.

Also considering we are hiring for code insights (both backend and frontend) and have a PM and EM lined up (Joel & Felix) and internally in the Web team we are already starting to treat it as a separate team, I don't think it's that premature to have a team page up.

Reply:
Agree with @jeanduplessis - this is a team that is set up well and I think it makes sense to start to separate some of the team info.

Reply:
I am fine with this if you think this is the best candidate experience for hiring. 

I suspect the candidate's concerns could also be addressed by simply organizing the information on the web team page better (which should probably happen regardless).

I want to ensure we don't prematurely fragment a single team in our documentation or in practice. Can you please chat with @lguychard to understand how he is approaching the same problem on the search team.

Reply:
@nicksnyder Felix and I chatted with Loïc and we are confident that we should move forward with a separate page.

Forming the Code Insights team is not something that is far off in our roadmap, in fact, Joel and Felix are actively working on this as their primary focus. To make this more explicit I will update the Web team page to indicate that Joel is the acting PM for the team until we hire a new dedicated PM.

Code Insights also differs from Web Platform and Extensions & Integrations in that it is much more of a classical product engineering team and as such, I think it is good to separate it out from the rest of the Web team in terms of how we position it to candidates.

We will also move ahead and set up a dedicated frontend engineer job spec for this team and have a separate hiring pipeline from the Web team for it, lead by Felix.

Reply:
So to be clear, are we ready to declare the code insights team as a separate team that has an EM (Felix) and a PM (Joel) where we are going to be actively hiring for? If all those are true, then let's lean in to this and be more explicit. Make this look like a complete team page, just like any other team.

Reply:
And also
- Update the growth plan on the web team page to pull out code insights since this PR is doing exactly that split already.
- Remove code insights goals from web team


If any of the above feels uncomfortable, then we aren't ready to document a new team yet and we should instead focus in improving the existing web team's docs to solve the candidate problem.

Reply:
Please review this: https://github.com/sourcegraph/about/pull/2153

Reply:
I'll let others comment, but I'm comfortable splitting code insights out into its own handbook pages at this point, and think that may help clear up some confusion with the page organization. I'll leave some other comments on #2153 about the web team's overall focus.

Reply:
I agree we should lean in and commit to the formation of this team and create a distinction from the web team. 

@felixfbecker @Joelkw let's update this PR so the new page has all the relevant info needed.